ocean_py/asset/asset_off_chain.py

```diff
@@ -0,0 +1,81 @@
+"""
+    Asset class to hold Ocean asset information such as asset id and metadata
+"""
+import re
+from web3 import Web3
+
+from squid_py.did import (
+    did_parse,
+    id_to_did,
+)
+
+from ocean_py.asset.asset_base import AssetBase
+# from ocean_py import logger
+
+class AssetOffChain(AssetBase):
+    def __init__(self, ocean, did=None, agent=None, **kwargs):
+        """
+        init an asset class with the following:
+        :param ocean: ocean object to use to connect to the ocean network
+        :param did: Optional did of the asset
+        """
+        AssetBase.__init__(self, ocean, did)
+
+        if agent:
+            self._agent = agent
+
+        if self._did:
+            # look for did:op:xxxx/yyy, where xxx is the agent and yyy is the asset id
+            data = did_parse(self._did)
+            if data['id_hex'] and data['path']:
+                agent_did = id_to_did(data['id_hex'])
+                if not self._agent:
+                    self._agent = ocean.resolve_agent(agent_did, **kwargs)
+                self._id = re.sub(r'^0[xX]', '', Web3.toHex(hexstr=data['path']))
```

Comment:
Instead of doing it as a regex substition should we use the function that did.py uses?
https://eth-utils.readthedocs.io/en/latest/utilities.html?highlight=remove_0x_prefix#remove-0x-prefix-value-string
-or-
Does that function NOT handle '0X' -- in which case should ocean-py have a utility library function called remove-0x-prefix that does the right thing?

Reply:
I originally I was using this function, but I'm trying to remove any extra dependencies to the library as much as possible. But maybe I should put it back since we are already using web3